token/js/test/e2e-2022/transferHook.test.ts

```diff
@@ -15,31 +15,57 @@ import {
     updateTransferHook,
     AuthorityType,
     setAuthority,
+    createAssociatedTokenAccountInstruction,
+    getAssociatedTokenAddressSync,
+    ASSOCIATED_TOKEN_PROGRAM_ID,
+    createMintToCheckedInstruction,
+    createAssociatedTokenAccountIdempotentInstruction,
+    createTransferCheckedInstruction,
+    getExtraAccountMetaAccount,
+    ExtraAccountMetaListLayout,
+    ExtraAccountMetaLayout,
+    addExtraAccountsToInstruction,
 } from '../../src';
-import { TEST_PROGRAM_ID, newAccountWithLamports, getConnection } from '../common';
+import { TEST_PROGRAM_ID, newAccountWithLamports, getConnection, TRANSFER_HOOK_TEST_PROGRAM_ID } from '../common';
+import { createHash } from 'crypto';
 
 const TEST_TOKEN_DECIMALS = 2;
 const EXTENSIONS = [ExtensionType.TransferHook];
 describe('transferHook', () => {
     let connection: Connection;
     let payer: Signer;
+    let payerAta: PublicKey;
+    let destinationAuthority: PublicKey;
+    let destinationAta: PublicKey;
     let transferHookAuthority: Keypair;
+    let pdaExtraAccountMeta: PublicKey;
     let mint: PublicKey;
-    let transferHookProgramId: PublicKey;
-    let newTransferHookProgramId: PublicKey;
     before(async () => {
         connection = await getConnection();
         payer = await newAccountWithLamports(connection, 1000000000);
+        destinationAuthority = Keypair.generate().publicKey;
         transferHookAuthority = Keypair.generate();
-        transferHookProgramId = Keypair.generate().publicKey;
-        newTransferHookProgramId = Keypair.generate().publicKey;
     });
     beforeEach(async () => {
         const mintKeypair = Keypair.generate();
         mint = mintKeypair.publicKey;
+        pdaExtraAccountMeta = getExtraAccountMetaAccount(TRANSFER_HOOK_TEST_PROGRAM_ID, mint);
```

Comment:
nit: This can be done separately, but I'm realizing now that this method isn't actually fetching an account, it's deriving an address, so it should be called `getExtraAccountMetaAddress`, and we should reverse the arguments for consistency with the address deriving functions

Reply:
Will create a new PR for that!